### PR TITLE
(cloud-merge) Fix the cast failed when RollupJobV2 convert to CloudRollupJobV2

### PR DESCRIPTION
**Pull request by Lchangliang**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Lchangliang** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Review by w41ter — APPROVED** [review]

**Comment by Lchangliang** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/alter/CloudRollupJobV2.java [hunk 1]
```diff
@@ -41,29 +41,35 @@
...
 
-    public static AlterJobV2 buildCloudRollupJobV2(RollupJobV2 job) throws IOException {
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        DataOutputStream dos = new DataOutputStream(baos);
-        job.write(dos);
-        ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
-        DataInputStream dis = new DataInputStream(bais);
-        CloudRollupJobV2 ret = (CloudRollupJobV2) CloudRollupJobV2.read(dis);
-        ret.partitionIdToRollupIndex = job.partitionIdToRollupIndex;
+    public static AlterJobV2 buildCloudRollupJobV2(RollupJobV2 job) throws IllegalAccessException {
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/alter/CloudRollupJobV2.java:58]: consider  using template in the future.

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**